packages/aws-cdk/lib/version.ts

```diff
@@ -97,19 +91,30 @@ export async function latestVersionIfHigher(currentVersion: string, cacheFile: V
   }
 }
 
-export async function displayVersionMessage(): Promise<void> {
+function getMajorVersionUpgradeMessage(currentVersion: string): string | void {
+  const currentMajorVersion = semver.major(currentVersion);
+  if (UPGRADE_DOCUMENTATION_LINKS[currentMajorVersion]) {
+    return `Information about upgrading from version ${currentMajorVersion}.x to version ${currentMajorVersion + 1}.x is available here: ${UPGRADE_DOCUMENTATION_LINKS[currentMajorVersion]}`;
+  }
+}
+
+function getVersionMessage(currentVersion: string, laterVersion: string): string[] {
+  return [
+    `Newer version of CDK is available [${colors.green(laterVersion as string)}]`,
+    getMajorVersionUpgradeMessage(currentVersion),
+    'Upgrade recommended (npm install -g aws-cdk)',
+  ].filter(Boolean) as string[];
+}
+
+export async function displayVersionMessage(currentVersion = versionNumber(), versionCheckCache?: VersionCheckTTL): Promise<void> {
   if (!process.stdout.isTTY || process.env.CDK_DISABLE_VERSION_CHECK) {
     return;
   }
 
   try {
-    const versionCheckCache = new VersionCheckTTL();
-    const laterVersion = await latestVersionIfHigher(versionNumber(), versionCheckCache);
+    const laterVersion = await latestVersionIfHigher(currentVersion, versionCheckCache ?? new VersionCheckTTL());
```

Comment:
I added the ability to pass a version check file via dependency injection to this method to add additional tests. This ensures that the tests aren't mucking with the global version check file.